### ISSUE TITLE
Milestones not sorted by name in dropdown

### ISSUE THREAD

**Issue by stuzer05** [go-gitea/gitea#31674]
### Gitea Version

1.23.0+dev-332-gba9589a92d

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![image](https://github.com/user-attachments/assets/8c2afc9f-496f-495b-94b4-8915752f6f39)

### How are you running Gitea?

docker

### Database

PostgreSQL

**Comment by yp05327** [comment]
By default, it is sorted by name. But for the `order` in SQL, if there are numbers in it, they will be handled as characters.

**Comment by stuzer05** [comment]
> By default, it is sorted by name. But for the `order` in SQL, if there are numbers in it, they will be handled as characters.

There're workarounds to achieve natural order, such as 
```sql
SELECT * FROM milestone WHERE repo_id = 40 ORDER BY LENGTH(name), name asc
```

![image](https://github.com/user-attachments/assets/f628c524-fad3-4777-a2e6-dec0a708e7f3)

**Comment by yp05327** [comment]
Yes, `LENGTH` can fix this, but I'm not sure whether this function can be used in all DB.
And what about other names. e.g. project name, issue's title, organization/user name...

**Comment by stuzer05** [comment]
In most cases alphabetical sorting would be best because that's the way we look. It's best to set it by default and use another sorting for specific cases

**Comment by stuzer05** [comment]
|     RDBMS     | Supported? |                   Notes                  |
|:-------------:|:----------:|:----------------------------------------:|
| MySQL/MariaDB | Yes        |                                          |
| PostgreSQL    | Yes        |                                          |
| Oracle        | Yes        |                                          |
| SQL Server    | Yes        | Uses LEN(field) instead of LENGTH(field) |
| SQLite        | Yes        |                                          |

**Comment by lunny** [comment]
I know we can do that with SQL but I don't think the new order method can satisfy every situation. For example, assume the milestone names are `A1111`, `B22`, and `C3`. What's the right order do you think?

**Comment by stuzer05** [comment]
> A1111

Exactly as you typed: `A1111`, `B22`, `C3`. In this case it's grouped by "prefix". But here can be another meanings, I know
